viewing_party/party.py

```diff
@@ -73,111 +55,100 @@ def get_most_watched_genre(user_data):
-    for movie in user_watched:
-        if movie not in friend_watched:
-            user_unique_watched.append(movie)
+        if movie["title"] not in friend_watched_titles:
+            user_unique_watched_movies.append(movie)
 
-    return user_unique_watched    
+    return user_unique_watched_movies
 
 def get_friends_unique_watched(user_data):
-    user_watched = []
-    friend_watched = []  
-    friends_unique_watched = []
-
-    for watched_movies in user_data['friends']:
-        for movie in watched_movies["watched"]:
-            if movie not in friend_watched:
-                friend_watched.append(movie)
-    for movie in user_data["watched"]:
-        user_watched.append(movie)
-    for movie in friend_watched:
-        if movie not in user_watched:
-            friends_unique_watched.append(movie)
+    user_watched_titles = get_movie_titles(user_data["watched"])
+    friends_movies = get_friends_movies(user_data)
+    friends_watched_titles = set()
+    friends_unique_watched_movies = []
+
+    for movie in friends_movies:
+        title = movie["title"]
+        if title not in friends_watched_titles and title not in user_watched_titles:
+            friends_watched_titles.add(title)
+            friends_unique_watched_movies.append(movie)
 
-    return friends_unique_watched    
+    return friends_unique_watched_movies
 
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 def get_available_recs(user_data):
-    user_list = user_data["watched"]
-    friends_list = user_data["friends"]
-
     if "subscriptions" not in user_data or not user_data["subscriptions"]:
-        subscriptions = set([])
-    else:
-        subscriptions = set(user_data["subscriptions"])
-
-    user_set = set()
+        return []
+    
+    friends_movie_list = get_friends_unique_watched(user_data)
+    subscriptions = set(user_data["subscriptions"])
+    user_watched_titles = get_movie_titles(user_data["watched"])
+    recommended_movies = []
     recommended_set = set()
-    recommended = []
     
-    for movie in user_list:
-        user_set.add(movie['title'])
-        
-    for friend in friends_list:
-        if friend['watched']:
-            for movie in friend['watched']:
-                if movie['host'] in subscriptions:
-                    movie_title = movie['title']
-                    if movie_title not in user_set and movie_title not in recommended_set:
-                        recommended.append(movie)
-                        recommended_set.add(movie_title)
-
-    return recommended
+    for movie in friends_movie_list:
+        if movie['host'] in subscriptions:
+            movie_title = movie['title']
+            if movie_title not in user_watched_titles and movie_title not in recommended_set:
+                recommended_movies.append(movie)
+                recommended_set.add(movie_title)
+
+    return recommended_movies

```

Comment:
This function still has some repeated logic. 

You invoke `get_friends_unique_watched`. This function returns a list of movies that at least one of the user's friends have watched, but the user has not watched. 

Therefore you do not need to get a list of movies a user has watched and then compare them against `friends_movie_list`. 

`get_available_recs` can be written like this:

```py
def get_available_recs(user_data):
    friends_unique_movies = get_friends_unique_watched(user_data)
    subscriptions = set(user_data["subscriptions"])

    return [movie for movie in friends_unique_movies if movie["host"] in subscriptions]
```